src/types.rs

```diff
@@ -212,12 +212,15 @@ impl Decodable for PublicKey {
   }
 }
 
+// Note: name field is initially same as original_name, this should be replaced by relocated name
+// calculated by the nodes close to original_name by using assign_relocated_name method
 #[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
 pub struct PublicId {
   pub public_key: PublicKey,
   pub public_sign_key: PublicSignKey,
   pub validation_token: Signature,
-  pub name: NameType
+  name: NameType,
+  relocated: bool,
```

Comment:
I would favour to not have a bool at all, simply the name field.  either it corresponds to the calculated result on the other fields, or it doesn't.  If it doesn't, it is only valid if it is already stored in a routing table of a close group.  This only works because vaults are non-persistent.